### PR TITLE
Fix GriefDefender Hook

### PR DESCRIPTION
**Pull request by ahdg6**
Solved https://github.com/Ste3et/FurnitureLib/issues/70

### CODE REVIEW

**Comment by hkkongou** [comment]
> Solved [Ste3et/FurnitureLib#70](https://github.com/Ste3et/FurnitureLib/issues/70)

thank you so much dalao

**Comment by ahdg6** [comment]
我编译不了=-=，得等作者自己看看能不能解决你的问题，除此之外我还添加了一个自定义Flag允许玩家开关保护
图片
如果可以的话帮忙解释一下我因为缺少依赖无法编译，所以还没测试过，谢啦(￣Д￣)ﾉ

File: Core/src/main/java/de/Ste3et_C0st/ProtectionLib/main/plugins/fGriefdefenderAPI.java [hunk 1]
```diff
@@ -9,20 +9,32 @@
...
 public class fGriefdefenderAPI extends ProtectionConfig implements Listener {
+	private final Flag PROTECTIONLIB_BUILD_PROTECT;
 
 	public fGriefdefenderAPI(Plugin plugin) {
 		super(plugin);
 		Bukkit.getPluginManager().registerEvents(this, ProtectionLib.getInstance());
+
+        PROTECTIONLIB_BUILD_PROTECT = Flag.builder()
+				.id("protectionlib:build_protect")
+				.name("protectionlib-build-protect")
```
Comment (bloodmc) [inline Core/src/main/java/de/Ste3et_C0st/ProtectionLib/main/plugins/fGriefdefenderAPI.java]: Should be `build-protect`

File: Core/src/main/java/de/Ste3et_C0st/ProtectionLib/main/plugins/fGriefdefenderAPI.java [hunk 2]
```diff
@@ -52,18 +64,21 @@ public boolean canBuild(Player player, Location loc) {
 		ClaimManager claimManager = GriefDefender.getCore().getClaimManager(loc.getWorld().getUID());
 		if(Objects.isNull(claimManager)) return true; 
 		Claim claim = claimManager.getClaimAt(loc.getBlockX(), loc.getBlockY(), loc.getBlockZ());
-		if(Objects.nonNull(claim) == true) {
-			 return claim.isTrusted(player.getUniqueId());
+		final Subject subject = GriefDefender.getCore().getSubject(player.getUniqueId().toString());
+		final Tristate useProtect = claim.getActiveFlagPermissionValue(PROTECTIONLIB_BUILD_PROTECT, subject, null, true);
+		final Tristate canBuild = claim.getActiveFlagPermissionValue(com.griefdefender.api.permission.flag.Flags.BLOCK_PLACE, subject, null, true);
```
Comment (bloodmc) [inline Core/src/main/java/de/Ste3et_C0st/ProtectionLib/main/plugins/fGriefdefenderAPI.java]: Should be 
`final Tristate canBuild = GriefDefender.getPermissionManager().getActiveFlagPermissionValue(claim, subject, com.griefdefender.api.permission.flag.Flags.BLOCK_PLACE, player, location.getBlock(), null, TrustTypes.BUILDER, true);`

File: Core/src/main/java/de/Ste3et_C0st/ProtectionLib/main/plugins/fGriefdefenderAPI.java [hunk 3]
```diff
@@ -9,20 +9,32 @@
...
 import de.Ste3et_C0st.ProtectionLib.main.ProtectionLib;
 
 public class fGriefdefenderAPI extends ProtectionConfig implements Listener {
+	private final Flag PROTECTIONLIB_BUILD_PROTECT;
 
 	public fGriefdefenderAPI(Plugin plugin) {
 		super(plugin);
 		Bukkit.getPluginManager().registerEvents(this, ProtectionLib.getInstance());
+
+        PROTECTIONLIB_BUILD_PROTECT = Flag.builder()
```
Comment (bloodmc) [inline Core/src/main/java/de/Ste3et_C0st/ProtectionLib/main/plugins/fGriefdefenderAPI.java:33]: Why is this custom flag needed? This plugin can just make use of `Flags.BLOCK_PLACE` assuming the only check is doing is on block place.

**Pull request merged**